packages/ui-patterns/AssistantChat/AssistantChatForm.tsx

```diff
@@ -0,0 +1,148 @@
+import { Loader2 } from 'lucide-react'
+import React, { ChangeEvent, createRef, useEffect } from 'react'
+import { TextArea } from 'ui/src/components/shadcn/ui/text-area'
+import { cn } from 'ui/src/lib/utils'
+
+export interface FormProps extends React.FormHTMLAttributes<HTMLFormElement> {
+  /* The ref for the textarea */
+  textAreaRef: React.RefObject<HTMLTextAreaElement>
+  /* The loading state of the form */
+  loading: boolean
+  /* The disabled state of the form */
+  disabled?: boolean
+  /* The value of the textarea */
+  value?: string
+  /* The function to handle the value change */
+  onValueChange: (value: ChangeEvent<HTMLTextAreaElement>) => void
+  /* Used to stop onSubmit event when Command popover is open. Use with AssistantCommandsPopover */
+  commandsOpen?: boolean
+  /* Used to close Command popover when onSubmit event happens. Use with AssistantCommandsPopover */
+  setCommandsOpen?: (value: boolean) => void
+  /* The icon to display on the left of the textarea */
+  icon?: React.ReactNode
+  /* The function to handle the form submission */
+  onSubmit: React.FormHTMLAttributes<HTMLFormElement>['onSubmit']
+}
+
+const AssistantChatForm = React.forwardRef<HTMLFormElement, FormProps>(
+  (
+    {
+      loading = false,
+      disabled = false,
+      value = '',
+      textAreaRef,
+      commandsOpen = false,
+      icon = null,
+      onValueChange,
+      setCommandsOpen,
+      onSubmit,
+      ...props
+    },
+    ref
+  ) => {
+    const formRef = createRef<HTMLFormElement>()
+    const submitRef = createRef<HTMLButtonElement>()
+
+    /**
+     * This effect is used to resize the textarea based on the content
+     */
+    useEffect(() => {
+      if (textAreaRef) {
+        if (!value && textAreaRef && textAreaRef.current) {
+          textAreaRef.current.style.height = '40px'
+        } else if (textAreaRef && textAreaRef.current) {
+          const newHeight = textAreaRef.current.scrollHeight + 'px'
```

Comment:
Does there need to be a way to set max height on this? I didn't build it to check, but from reading this, it seems like it might be able to grow all the way off the screen

Reply:
ahh yes! probably!

I'll merge this in anyway, it's not used in any apps yet so we can do a follow up PR integrating it into an app.